bindings/flow/Tuple.h

```diff
@@ -68,13 +69,14 @@ struct Tuple {
 		return append(t);
 	}
 
-	enum ElementType { NULL_TYPE, INT, BYTES, UTF8, BOOL, FLOAT, DOUBLE, UUID, NESTED };
+	enum ElementType { NULL_TYPE, INT, BYTES, UTF8, BOOL, FLOAT, DOUBLE, UUID, NESTED, VERSIONSTAMP };
```

Comment:
It's a little concerning to me that the `Tuple` code in the flow bindings and within fdbclient are continuing to diverge. In this case in particular, it's concerning because we have a bunch of tests around tuple packing/unpacking in the binding tester, including compatibility testing with different bindings, but they don't help us validate the `fdbclient` Tuple implementation because the code bases are different.

In theory, I think we could do something like add a new folder with the `Tuple` implementation, and then the build system could add that to both the flow bindings and to fdbclient and server if we wanted to share code. I'm not sure that we should block fdbserver/fdbclient from having the new type if we want it for, say, `get_mapped_range` support, but maybe it's something we should consider doing in the future.

In particular, there are still a few types (namely `NESTED` and `UUID`) that exist in the flow bindings (and other bindings) but not in fdbclient.

Reply:
makes sense, do you think I should resolve this divergence in this change or I can leave it as a separate future change?

Reply:
I guess it depends on how difficult we think it would be. Perhaps @ammolitor might have an opinion on the level of effort to combine the two implementations and then share code, especially if we care about having the header files of the flow bindings be backwards compatible (if we care about not breaking flow bindings users)

Reply:
Will discuss offline, it might be a good idea we have a separate PR for this change, since it is not a blocker of this change.

Reply:
Will discuss offline, it might be a good idea we have a separate PR for this change, since it is not a blocker of this change.

Reply:
FYI: there is already an issue on this #4351.